### PR TITLE
52 languages multilingual recipe (for espnet2)

### PR DESCRIPTION
**Pull request by ftshijt**
This PR is intended to extend our multilingual models to espnet2, consisting of Voxforge and commonvoice5.1 (52 languages)

### CODE REVIEW

File: egs2/TEMPLATE/asr1/asr.sh [hunk 1]
```diff
@@ -439,7 +439,7 @@ if ! "${skip_data_prep}"; then
                 else
                     _suf=""
                 fi
-                utils/copy_data_dir.sh data/"${dset}" "${data_feats}${_suf}/${dset}"
+                utils/copy_data_dir.sh --validate_opts --non-print data/"${dset}" "${data_feats}${_suf}/${dset}"
```
Comment (ftshijt) [inline egs2/TEMPLATE/asr1/asr.sh:442]: I add non-print here for some minor langs in commonvoice

**Comment by danpovey** [comment]
Just curious, how much benefit do you guys get from the multilingual models vs. training models individually?

**Comment by ftshijt** [comment]
> Just curious, how much benefit do you guys get from the multilingual models vs. training models individually?

This multilingual model, as a pre-trained model, generally works for minor languages without many resources. We have observed 11% relative improvements on WER. Please see [this](http://www.interspeech2020.org/uploadfile/pdf/Mon-3-1-3.pdf) for some details

**Comment by houwenxin** [comment]
> 
> 
> > Just curious, how much benefit do you guys get from the multilingual models vs. training models individually?
> 
> This multilingual model, as a pre-trained model, generally works for minor languages without many resources. We have observed 11% relative improvements on WER. Please see [this](http://www.interspeech2020.org/uploadfile/pdf/Mon-3-1-3.pdf) for some details

I would like to supplement one more point. For the multilingual model itself, [[1]](https://arxiv.org/pdf/2007.03001.pdf) shows that it helps improve the performance on low-resource languages in training data with a sacrifice of performance degradation on high-resource languages. And the gap can be reduced or even closed by increasing the model size and training data [[1]](https://arxiv.org/pdf/2007.03001.pdf).

**Comment by kamo-naoyuki** [comment]
Will you add the result? I'm curious

**Comment by sw005320** [comment]
@ftshijt, can you add some temporal results, if possible?
We can also add some TODO items here or a new issue.
Then, I'll merge this PR and we will make some follow up PRs.
I'll work on a couple of TODO items with you.

**Comment by ftshijt** [comment]
> @ftshijt, can you add some temporal results, if possible?
> We can also add some TODO items here or a new issue.
> Then, I'll merge this PR and we will make some follow up PRs.
> I'll work on a couple of TODO items with you.

Sure, I downloaded the model files yesterday (not converged but reasonable to have some tests) from the AWS server and are working on some low-resource languages results today. Ideally, I could come up with some results before Wednesday.

**Comment by ftshijt** [comment]
> Will you add the result? I'm curious

Yes, I'm working on it. Thanks for following up.

**Comment by ftshijt** [comment]
I turned back to the transformer as the conformer configuration seems not to work for the current recipe. This PR is ready to merge.

File: egs2/open_li52/asr1/local/data.sh [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+. ./db.sh || exit 1;
+
+# general configuration
+stage=0       # start from 0 if you need to start from data preparation
+stop_stage=100
+SECONDS=0
+langs="en de fr cy br cv ky ga-IE sl cnh et mn sah dv sv-SE id ar ta ia lv ja rm-sursilv hsb ro fy-NL el rm-vallader as mt ka or vi pa-IN tt kab ca zh-TW it fa eu es ru tr nl eo zh-CN rw pt zh-HK cs pl uk"
```
Comment (sw005320) [inline egs2/open_li52/asr1/local/data.sh:14]: This is too long.
Also, is it based on some order?
I want to sort the entry in this list with some criteria.

**Review by sw005320 — COMMENTED** [review]
Thanks. One quick comment.

File: egs2/open_li52/asr1/local/data.sh (hunk 2 above)
Comment (ftshijt) [reply egs2/open_li52/asr1/local/data.sh:14]: Sure, let's do it alphabetically.

**Comment by sw005320** [comment]
Thanks, @ftshijt!

**Pull request merged**